Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +60,135 @@ class TransactionHistoryViewController: UIViewController {
-            self.btnFilter.setTitle(LocalizedStrings.staking.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 5:
-            self.Filtercontent = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}
-            self.btnFilter.setTitle("Coinbase (".appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        default:
-            self.Filtercontent = self.mainContens
-            self.btnFilter.setTitle(LocalizedStrings.all.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-        }
-    }
-    
     func updateFilterDropdownItems() {
-        let sentCount = self.mainContens.filter{$0.Direction == 0}.count
-        let ReceiveCount = self.mainContens.filter{$0.Direction == 1}.count
-        let yourselfCount = self.mainContens.filter{$0.Direction == 2}.count
-        let stakeCount = self.mainContens.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
-        let coinbaseCount = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}.count
+        let sentCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionSent}.count
+        let receiveCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionReceived}.count
+        let yourselfCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionTransferred}.count
+        let stakeCount = self.allTransactions.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
+        let coinbaseCount = self.allTransactions.filter{$0.Type == DcrlibwalletTxTypeCoinBase}.count
         
         self.btnFilter.items.removeAll()
-        self.btnFilter.setTitle(LocalizedStrings.all.appending(" (").appending(String(self.mainContens.count)).appending(")"), for: .normal)
-        self.btnFilter.items.append(LocalizedStrings.all.appending(" (").appending(String(self.mainContens.count)).appending(")"))
+        self.btnFilter.setTitle(LocalizedStrings.all.appending("(\(self.allTransactions.count))"), for: .normal)
+        self.btnFilter.items.append(LocalizedStrings.all.appending("(\(self.allTransactions.count))"))
         
-        self.filtertitle.removeAll()
-        self.filtertitle.append(0)
+        self.filters.removeAll()
+        self.filters.append(DcrlibwalletTxFilterAll)
         
         if sentCount != 0 {
-            self.btnFilter.items.append(LocalizedStrings.sent.appending(" (").appending(String(sentCount)).appending(")"))
-            self.filtertitle.append(1)
+            self.btnFilter.items.append(LocalizedStrings.sent.appending("(\(sentCount))"))
+            self.filters.append(1)
```

Comment:
`append(DcrlibwalletTxFilterSent)`
Do similar for the other cases below.

Reply:
fixed in 28f46d1